iml-agent/src/action_plugins/high_availability/mod.rs

```diff
@@ -65,46 +73,10 @@ fn create(elem: &Element) -> ResourceAgentInfo {
     }
 }
 
-pub(crate) async fn crm_attribute(args: Vec<String>) -> Result<String, ImlAgentError> {
-    let o = Command::new("crm_attribute")
-        .args(args)
-        .checked_output()
-        .await?;
-
-    Ok(String::from_utf8_lossy(&o.stdout).to_string())
-}
-
-pub(crate) async fn crm_resource<I, S>(args: I) -> Result<String, ImlAgentError>
-where
-    I: IntoIterator<Item = S>,
-    S: AsRef<OsStr>,
-{
-    let o = Command::new("crm_resource")
-        .args(args)
-        .checked_output()
-        .await?;
-
-    Ok(String::from_utf8_lossy(&o.stdout).to_string())
-}
-
-pub(crate) async fn pcs(args: Vec<String>) -> Result<String, ImlAgentError> {
-    let o = Command::new("pcs").args(args).checked_output().await?;
-
-    Ok(String::from_utf8_lossy(&o.stdout).to_string())
-}
-
-async fn crm_mon_status() -> Result<Element, ImlAgentError> {
-    let o = Command::new("crm_mon")
-        .args(&["--as-xml", "--inactive"])
-        .checked_output()
-        .await?;
-
-    Ok(Element::from_reader(o.stdout.as_slice())?)
-}
-
-fn resource_status(tree: Element) -> HashMap<String, String> {
+fn resource_status(rawxml: &[u8]) -> Result<HashMap<String, String>, ImlAgentError> {
```

Comment:
This should be added to the `Cluster` struct generated by `read_crm_output` instead of parsing the raw cibxml.

We don't need two different ways to do the same thing, and using a streaming reader is more error tolerant than searching a document + find + unwrapping.

Reply:
`read_crm_output` is insufficient, and limited in scope like the python code, and these action plugins aren't self-limiting to certain resource agents.

Reply:
> read_crm_output is insufficient, and limited in scope like the python code

How? It can parse the entire cibxml.

Reply:
It does a similar but smaller check than `_get_resource_locations` in the python code only returning certain resource types, and it doesn't do any accounting for cloned or grouped resources.
```rust
fn is_lustre_ra(x: &str) -> bool {
    x == "ocf::ddn:lustre-server" || x == "ocf::lustre:Lustre"
}
...
                if x.get("resource_agent")
                    .filter(|x| is_lustre_ra(x))
                    .is_none()
                {
                    continue;
                }
```
If I change that, than what gets reported by the corosync daemon plugin changes.

Reply:
I think it would be better to expand the `Cluster` struct to account for all resource types + clones and groups and then filter downstream in the places we need just Lustre resources, rather than have two ways of doing mostly the same thing.